src/config.py

```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Manager for handling Kafka configuration."""
+
+import logging
+from typing import Dict, List, Optional
+
+from charms.kafka.v0.kafka_snap import SNAP_CONFIG_PATH
+from ops.charm import CharmBase
+from ops.model import Unit
+
+from literals import PEER, REL_NAME, ZK
+from utils import safe_write_to_file
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_CONFIG_OPTIONS = """
+log.dirs=/var/snap/kafka/common/log
+clientPort=2181
+listeners=SASL_PLAINTEXT://:9092
+sasl.enabled.mechanisms=SCRAM-SHA-512
+sasl.mechanism.inter.broker.protocol=SCRAM-SHA-512
+security.inter.broker.protocol=SASL_PLAINTEXT
+security.protocol=SASL_PLAINTEXT
+authorizer.class.name=kafka.security.authorizer.AclAuthorizer
+allow.everyone.if.no.acl.found=false
+listener.name.sasl_plaintext.sasl.enabled.mechanisms=SCRAM-SHA-512
+"""
+
+
+class KafkaConfig:
+    """Manager for handling Kafka configuration."""
+
+    def __init__(self, charm: CharmBase):
+        self.charm = charm
+        self.default_config_path = SNAP_CONFIG_PATH
+        self.properties_filepath = f"{self.default_config_path}/server.properties"
+        self.jaas_filepath = f"{self.default_config_path}/kafka-jaas.cfg"
+
+    @property
+    def sync_password(self) -> Optional[str]:
+        """Returns charm-set sync_password for server-server auth between brokers."""
+        return self.charm.model.get_relation(PEER).data[self.charm.app].get("sync_password", None)
+
+    @property
+    def zookeeper_config(self) -> Dict[str, str]:
+        """The config from current ZooKeeper relations for data necessary for broker connection.
+
+        Returns:
+            Dict of ZooKeeeper `username`, `password`, `endpoints`, `chroot`, `connect` and `uris`
```

Comment:
You are right, and well spotted. That said, Kafka cannot relate to multiple ZK applications.
I only loop through all relations so that I don't have to pass `event` to get the specific one.